lib/path.js

```diff
@@ -1,7 +1,14 @@
 'use strict';
 
+const util = require('util');
 const isWindows = process.platform === 'win32';
 
+function assertPath(path) {
+  if (typeof path !== 'string')
+    throw new TypeError('Path must be a string. Received ' +
+                        util.inspect(path));
```

Comment:
Minor style nit: can you put braces around the body here?  It doesn't fit on a single line.